src/SFA.DAS.Campaign.Web/Controllers/RegisterInterestController.cs

```diff
@@ -0,0 +1,45 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+using Microsoft.AspNetCore.Mvc;
+using SFA.DAS.Campaign.Domain.DataCollection;
+using SFA.DAS.Campaign.Models.DataCollection;
+using SFA.DAS.Campaign.Web.Constants;
+using SFA.DAS.Campaign.Web.Models;
+
+namespace SFA.DAS.Campaign.Web.Controllers
+{
+    public class RegisterInterestController : Controller
+    {
+        private readonly IUserDataCollection _userDataCollection;
+
+        public RegisterInterestController(IUserDataCollection userDataCollection)
+        {
+            _userDataCollection = userDataCollection == null ? throw new NullReferenceException() : userDataCollection ;
```

Comment:
From a code style point of view I personally don't see the value in this, it would fail before this if unable to resolve the dependency.